### PR TITLE
[v6r14] Added support for RFC proxies + reading VOMS extensions

### PR DESCRIPTION
**Pull request by acasajus**
NEW: Added support for RFC proxies throughout DIRAC
NEW: Read voms information in proxies directly without using voms-proxy-info
BACKPORT: Added LICENSE print as required by GPLv3
BACKPORT: Reloader. If the --autoreload flag is enabled, if there's a change in the python code the running process will automatically reload.

WARNING: Requires pyGSI 0.6.2 (set in externals commit 0509155)

### CODE REVIEW

File: Core/Security/VOMS.py [hunk 1]
```diff
@@ -8,8 +8,7 @@
 from DIRAC.Core.Security.BaseSecurity import BaseSecurity
 from DIRAC.Core.Security.X509Chain import X509Chain
 from DIRAC.Core.Utilities.Subprocess import shellCall
-from DIRAC.Core.Utilities import List
-from DIRAC.Core.Utilities import Os
+from DIRAC.Core.Utilities import List, Time, Os
```
Comment (sposs) [inline Core/Security/VOMS.py]: Comment: PEP8 discourages single-line multi imports

File: Core/Security/X509Chain.py [hunk 2]
```diff
@@ -153,6 +158,14 @@ def __getProxyExtensionList( self, diracGroup = False ):
                                           'critical, digitalSignature, keyEncipherment, dataEncipherment' ) )
     if diracGroup and type( diracGroup ) in self.__validExtensionValueTypes:
       extList.append( crypto.X509Extension( 'diracGroup', diracGroup ) )
+    if rfc or rfcLimited:
+      blob = [ [ "1.3.6.1.5.5.7.21.1" ] ] if not rfcLimited else [ [ "1.3.6.1.4.1.3536.1.1.1.9" ] ]
```
Comment (sposs) [inline Core/Security/X509Chain.py]: OMG ;-)

**Comment by sposs** [comment]
Review OK

File: Core/Security/VOMS.py (hunk 1 above)
Comment (acasajus) [inline Core/Security/VOMS.py]: Although we don't follow strictly pep8, check out https://www.python.org/dev/peps/pep-0008/#id17 :beers:

File: Core/Security/X509Chain.py (hunk 2 above)
Comment (acasajus) [inline Core/Security/X509Chain.py]: That's nothing. Check out the beauty that the ASN1 structure a certificate is:

https://github.com/DIRACGrid/DIRAC/pull/2272/files#diff-7d2aac6b2e0163417e13f662e1e2ae88R176

Comment (sposs) [inline Core/Security/X509Chain.py]: Yeah I saw... Seems very easy and friendly to work with it...

File: Core/Security/VOMS.py (hunk 1 above)
Comment (sposs) [inline Core/Security/VOMS.py]: Right, forgot about the from statement specifics...

File: ConfigurationSystem/Client/Helpers/CSGlobals.py [hunk 3]
```diff
@@ -26,26 +75,13 @@ def getCSExtensions():
     Return list of extensions registered in the CS
     They do not include DIRAC
   """
-  from DIRAC import gConfig
-  return gConfig.getValue( '/DIRAC/Extensions', [] )
+  return Extensions().getCSExtensions()
```
Comment (fstagni) [inline ConfigurationSystem/Client/Helpers/CSGlobals.py]: 2 questions:
- Is this new helper class specific for v6r14? 
- The fact that it's been developed, does it mean that should be used everywhere in the code where gConfig.getValue( '/DIRAC/Extensions', [] ) is used? (I have this case in couple place in LHCb)

Comment (acasajus) [inline ConfigurationSystem/Client/Helpers/CSGlobals.py]: It is a backport from v7. That will consistently return the extension name (with DIRAC at the end) and also check that the extension exists.

File: Core/Security/VOMS.py [hunk 4]
```diff
@@ -88,6 +87,73 @@ def getVOMSProxyFQAN( self, proxy ):
...
+
+        return S_OK( "\n".join( lines ) )
+      else:
+        return S_ERROR( "NOT IMP" )
+
+    finally:
+      if proxyDict[ 'tempFile' ]:
+        self._unlinkFiles( proxyDict[ 'tempFile' ] )
+
+  def OLDgetVOMSProxyInfo( self, proxy, option = False ):
```
Comment (fstagni) [inline Core/Security/VOMS.py]: I wonder if we need to keep this function at all, or just removing it. It's not going to be used anymore, right?

File: Core/Security/VOMS.py [hunk 5]
```diff
@@ -88,6 +87,73 @@ def getVOMSProxyFQAN( self, proxy ):
     return self.getVOMSProxyInfo( proxy, "fqan" )
 
   def getVOMSProxyInfo( self, proxy, option = False ):
```
Comment (fstagni) [inline Core/Security/VOMS.py]: Maybe just a little docstring? :)

File: Core/Security/VOMS.py (hunk 4 above)
Comment (acasajus) [inline Core/Security/VOMS.py]: Just for my mental health and to be able to check how things where before the change. I can add a comment with 

#TODO: DELETEME AFTER 01122015

File: Core/Security/VOMS.py (hunk 5 above)
Comment (acasajus) [inline Core/Security/VOMS.py]: Maybe, but only if you behave :)

**Comment by atsareg** [comment]
Merged into integration branch

File: Core/Utilities/Devloader.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+  def addStuffToClose( self, stuff ):
+    self.__stuffToClose.append( stuff )
+
+  @property
+  def enabled( self ):
+    return self.__enabled
+
+  def watchFile( self, fp ):
+    if os.path.isfile( fd ):
```
Comment (atsareg) [inline Core/Utilities/Devloader.py]: fp or fd ?

Comment (acasajus) [inline Core/Utilities/Devloader.py]: fp I guess

**Pull request closed without merging**